src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -531,6 +531,10 @@ private function hydrateObjectValue(mixed $value, string $className, bool $allow
                 throw new BadRequestHttpException(sprintf('The model path "%s" was sent an invalid data type "%s" for a date.', $propertyPathForError, get_debug_type($value)));
             }
 
+            if ('' === $value) {
+                return $allowsNull ? null : throw new BadRequestHttpException(sprintf('The model path "%s" was sent invalid date data "%s". Setting empty date value is not allowed, send a non-empty value or make the property nullable.', $propertyPathForError, $value, $dateFormat)) ;
+            }
+
```

Comment:
We can't do that this is a BC break. If we want to implement this fix we have to add an option to activate or not your fix and to have it not activate by default.